tests/scripts/rm_ml_api_tests.sh

```diff
@@ -0,0 +1,306 @@
+#!/bin/bash
+#
+# Copyright (c) 2020, 2021 Red Hat, IBM Corporation and others.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+##### Script for validating RM-ML APIs #####
+
+# Get the absolute path of current directory
+CURRENT_DIR="$(dirname "$(realpath "$0")")"
+SCRIPTS_DIR="${CURRENT_DIR}"
+
+# Source the common functions scripts
+. ${SCRIPTS_DIR}/python_api_constants.sh
+
+# Tests to validate the RM-ML apis
+function rm_ml_api_tests() {
+	start_time=$(get_date)
+	FAILED_CASES=()
+	TESTS_FAILED=0
+	TESTS_PASSED=0
+	TESTS=0
+	((TOTAL_TEST_SUITES++))
+	
+	rm_ml_api_tests=("rm_ml_post_experiment")
+	declare -A app_array=([galaxies]="3")
+	
+	# check if the test case is supported
+	if [ ! -z "${testcase}" ]; then
+		check_test_case "rm_ml_api"
+	fi
+	
+	# create the result directory for given testsuite
+	echo ""
+	TEST_SUITE_DIR="${RESULTS}/rm_ml_api_tests"
+	mkdir -p ${TEST_SUITE_DIR}
+	
+	# If testcase is not specified run all tests	
+	if [ -z "${testcase}" ]; then
+		testtorun=("${rm_ml_api_tests[@]}")
+	else
+		testtorun=${testcase}
+	fi
+
+	for test in "${testtorun[@]}"
+	do
+		TEST_DIR="${TEST_SUITE_DIR}/${test}"
+		mkdir ${TEST_DIR}
+		SETUP="${TEST_DIR}/setup.log"
+		AUTOTUNE_LOG="${TEST_DIR}/${test}_autotune.log"
+		LOG="${TEST_SUITE_DIR}/${test}.log"
+
+		echo ""
+		echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~" | tee -a ${LOG}
+		echo "                    Running Test ${test}" | tee -a ${LOG}
+		echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"| tee -a ${LOG}
+
+		echo " " | tee -a ${LOG}
+		echo "Test description: ${rm_ml_api_test_description[$test]}" | tee -a ${LOG}
+		echo " " | tee -a ${LOG}
+
+		# create the autotune setup
```

Comment:
Remove autotune deploy, app deploy and get search space json as we will be mock the search space json in our tests.